ERCS/erc-x.md

```diff
@@ -0,0 +1,238 @@
+    }
+
+    function totalSupply() public view override returns (uint256) {
+        return assetBoundContract.totalSupply();
+    }
+
+    function balanceOf(
+        address owner
+    ) public view override(ERC721, IERC721) returns (uint256) {
+        return assetBoundContract.balanceOf(owner);
+    }
+
+    function approve(address, uint256) public pure override(ERC721, IERC721) {
+        revert("ABT: Approvals not allowed");
+    }
+
+    function setApprovalForAll(
+        address,
+        bool
+    ) public pure override(ERC721, IERC721) {
+        revert("ABT: Approvals not allowed");
+    }
+
+    function transferFrom(
+        address,
+        address,
+        uint256
+    ) public pure override(ERC721, IERC721) {
+        revert("ABT: Transfers not allowed");
+    }
+
+    function safeTransferFrom(
+        address,
+        address,
+        uint256
+    ) public pure override(ERC721, IERC721) {
+        revert("ABT: Transfers not allowed");
+    }
+
+    function safeTransferFrom(
+        address,
+        address,
+        uint256,
+        bytes memory
+    ) public pure override(ERC721, IERC721) {
+        revert("ABT: Transfers not allowed");
+    }
+}
+```
+
+### **Interface functions**
+
+The functions detailed below MUST be implemented.
+
+#### `ownerOf` function
+
+Returns the owner of the NFT specified by the `tokenId`. Will read from the `assetBoundContract` the owner and return it.
+
+```solidity
+function ownerOf(uint256 tokenId) external view returns (address);
+```
+
+#### `tokenExists` function
+
+Returns true if the owner of the token is not the zero address (`0x0`)
+
+```solidity
+function tokenExists(uint256 tokenId) external view returns (bool);
+```
+
+#### `totalSupply` function
+
+Gets the total amount of tokens stored by the assetBoundContract
+
+```solidity
+function totalSupply() external view returns (uint256);
+```
+
+#### `balanceOf` function
+
+Returns the number of NFTs in the assetBoundContract that an owner has.
+
+```solidity
+function balanceOf(address owner) external view returns (uint256);
+```
+
+## Rationale
+
+The design of PermaLink-ABTs centers around the goal of enabling permanent token binding while optimizing for gas efficiency, composability, and secure ownership structures. We chose to implement a `reveal` function in place of a traditional `mint` function to reduce gas costs and simplify on-chain state changes. Unlike minting, which creates tokens at runtime and incurs higher gas fees, the `reveal` function maps pre-allocated tokens stored in an array or mapping. This allows tokens to be activated on demand without the overhead of dynamic token creation. As a result, token issuers can prepare and store an entire supply in advance, with users later revealing and binding tokens when needed. This approach aligns with the use case of portfolio binding and asset hierarchies, where large numbers of tokens may need to be activated and bound efficiently. 
```

Comment:
Your specification doesn't define a `reveal` function. That said, we generally recommend _not_ specifying `mint`/`burn`-type functions because they're usually admin-only. The creator of a contract will know how to call the admin functions, so they don't need to be part of the standard (unless you expect it to be common for _anyone_ to be minting/burning/revealing).